crates/webidl/src/first_pass.rs

```diff
@@ -71,7 +73,7 @@ pub(crate) struct DictionaryData<'src> {
 
 #[derive(PartialEq, Eq, PartialOrd, Ord, Debug, Clone, Copy)]
 pub(crate) enum OperationId<'src> {
-    Constructor,
+    Constructor(IgnoreTraits<&'src str>),
```

Comment:
This looks a bit confusing, especially with `IgnoreTraits`. `OperationId` meant to be an identified of operation, not a storage of data about the operation. Is it possible to move the JS class name to the `OperationData`?

Reply:
I was originally thinking that yeah but ended up feeling that it was best here because it only makes sense in the context of a constructor, not for other operations. I'd be fine either way though!

Reply:
Perhaps! I'm gonna go ahead this PR and leave this for a future cleanup, but `self_name` in the original `first_pass_operation` method is always the same for constructors so there's no clear entry point for this special case with renamed constructors. We can always continue to iterate on this in tree though!